examples/nlp/text_classification/text_classification_with_bert.py

```diff
@@ -76,31 +78,45 @@
     add_time_to_log_dir=True,
 )
 
-""" Load the pretrained BERT parameters
-See the list of pretrained models, call:
-nemo_nlp.huggingface.BERT.list_pretrained_models()
-"""
+model_type = args.pretrained_model_name.split('-')[0]
+model_cls = nemo_nlp.utils.DEFAULT_MODELS[model_type]['class']
 
-if args.bert_checkpoint and args.bert_config:
-    pretrained_bert_model = nemo.collections.nlp.nm.trainables.common.huggingface.BERT(
-        config_filename=args.bert_config
-    )
-    pretrained_bert_model.restore_from(args.bert_checkpoint)
+if args.bert_config is not None:
+    model = model_cls(config_filename=args.bert_config)
 else:
-    pretrained_bert_model = nemo.collections.nlp.nm.trainables.common.huggingface.BERT(
-        pretrained_model_name=args.pretrained_bert_model
-    )
-
-hidden_size = pretrained_bert_model.hidden_size
-tokenizer = BertTokenizer.from_pretrained(args.pretrained_bert_model)
+    """ Use this if you're using a standard BERT model.
+    To see the list of pretrained models, call:
+    nemo_nlp.nm.trainables.huggingface.BERT.list_pretrained_models()
+    nemo_nlp.nm.trainables.huggingface.Albert.list_pretrained_models()
+    nemo_nlp.nm.trainables.huggingface.Roberta.list_pretrained_models()
+    """
+    model = model_cls(pretrained_model_name=args.pretrained_model_name)
+
+hidden_size = model.hidden_size
+
+tokenizer_cls = nemo_nlp.data.NemoBertTokenizer
+tokenizer_special_tokens = nemo_nlp.utils.MODEL_SPECIAL_TOKENS[model_type]
```

Comment:
How about moving MODEL_SPECIAL_TOKENS into the tokenizer and pass the model_type to it? @yzhang123

Reply:
done